src/PhotometryFit.cc

```diff
@@ -343,23 +344,33 @@ int PhotometryFit::minimize(const std::string &whatToFit, double nSigmaCut) {
 
     Eigen::SimplicialLDLT<SpMat> chol(hessian);
     // CholmodSimplicialLDLT2<SpMat> chol(hessian);
+
+    if (chol.info() != Eigen::Success) {
+        LOGLS_ERROR(_log, "minimize: factorization failed ");
+        return 2;
+    }
+
+    double old_chi2 = computeChi2().chi2;
+
+    Eigen::VectorXd delta = chol.solve(grad);
+    offsetParams(delta);
+    Chi2Statistic current_chi2(computeChi2());
+    LOGLS_DEBUG(_log, current_chi2);
+    if (current_chi2.chi2 > old_chi2) {
+        LOGL_WARN(_log, "chi2 went up, exiting outlier rejection loop");
+        returnCode = 1;
+        return returnCode;
+    }
+
```

Comment:
You should be able to use trace statements and leave them in; that's what trace was designed for.